Adds two tracks (#1)
kensho-technologies/career-tracks: individual_site_reliability_engineering.md

| **Technical Problem Solving** | Independently develops features scoped to entire application components; may involve multiple releases. Independently migrate/launch a new service in production. (e.g. SRE Migration of a service which include S3, ALB, traffic splits, envoy, alerts, etc). Understands how models are created, updated and used. |
| **Architecture** | Evaluates technology choices within a specific application component; considers tradeoffs between complexity, memory usage, performance, and speed of implementation.|
| **Implementation and Forethought** | Implementations perform as specified under typical conditions. Abnormal behavior from other components of the project is managed and resolved. Observables capture the state of the application. Independently provisions all dependencies for a service (e.g. Database, Load Balancer, ingress, dashboards, etc)  |
| **Code/Design Review** | Provides thorough, timely code reviews. Provides feedback on design docs about specific areas of expertise. |

> |**Collaboration**|Can evaluate the customer reliability of the service (SLO, SLA, SLI, etc). Proficient with continuous maturity model. Conducts blameless postmortems.|

Reviewer: ```suggestion
|**Collaboration**| Can evaluate the customer reliability of the service (SLO, SLA, SLI, etc). Proficient with continuous maturity model. Conducts blameless postmortems.|
```

| **Architecture** | Evaluate tradeoffs between complexity, memory usage, performance, reliability, and suitability for product objectives. |
| **Implementation & Forethought** | Abnormal behavior from dependencies is handled when possible; observables highlight actionable problems with the system. Scope is primarily within own project. Observability and migration expert (e.g. alerts/graphs). Identifies features and implements to reduce toil. Implements tools to track SLI, error budgets, mean time * metrics |
| **Code/Design Review** | Provides thorough, timely code reviews. Sought out as a reviewer for more complicated changes affecting an application. Provides feedback on all areas of design docs. |

> | **Collaboration** | Suggests improvements/enhancements to the technologies and also introduces better/advanced technology. Covers application dependencies and also monitoring/alerting tools. Train L1 folks and delegate more responsibilities. Train fellow SREs on handling pages (wheel-of-misfortune, etc.) Ability to convince stakeholders about the performance of our services and build trust and confidence. |

Reviewer: ```suggestion
| **Collaboration** | Suggests improvements/enhancements to the technologies and also introduces better/advanced technology. Covers application dependencies and also monitoring/alerting tools. Trains L1 folks and delegate more responsibilities. Trains fellow SREs on handling pages (wheel-of-misfortune, etc.) Ability to convince stakeholders about the performance of our services and build trust and confidence. |
```

<a name="ssre2"></a>
## Senior Site Reliability Engineer II

<a name="ssre2_techexc"></a>
### Technical Excellence
###### [Jump to top](#main)
|||
|---|---|
| **Tools and Tech** | Is sought out / seen as an expert in essential technologies for the role. Provides guidance regarding best practices. Deep experience with all essential technologies involved in production. Independently identifies the future needs and follows up with implementation. Evaluates and articulates the impacts of tradeoffs in complexity, memory usage, performance, reliability, and suitability for product goals across projects. Automates technology for easy/no effort adaption. (e.g. cluster-wide metrics and alerts). |
| **Technical Problem Solving**|Comes up with innovative ways to automate operational work. Understands model validation and graduation process (e.g. Graduate a model from stage to production). Improves the efficiency of our resources (e.g. Nodes/replica usage percentage, efficient and automated upgrades).|
| **Code/Design Review** | Provides guidance to colleagues in their system designs. |

> | **Collaboration** | Development of customized SRE policies and guidelines. (e.g. Database migration/upgrades, app migrations, model updates, caching of models/images). Work with applications and advocate to adapt common technologies. (e.g. Testing/validation methodologies). |

Reviewer: ```suggestion
| **Collaboration** | Development of customized SRE policies and guidelines. (e.g. Database migration/upgrades, app migrations, model updates, caching of models/images). Work with applications and advocate to adapt common technologies (e.g. testing/validation methodologies). |
```

| **Debugging & Support** |  Is sought out by teammates to help debug issues in products with which they have familiarity even with incomplete context. Offers mitigating or partial solutions when appropriate. Recognize common patterns of problems. Create a plan for resolution. Automated testing of disaster recovery scenarios. Automated gathering of data for issues. |

<a name="ssre2_leadership"></a>
### Leadership
###### [Jump to top](#main)
|||
|---|---|
| **Innovation** | Assesses associated risk of failure and learning trade-offs, and seeks to fail and learn frequently and rapidly to ultimately achieve desired results. *Frequently partners with junior teammates to increase failure/learning opportunities.* |

> | **Collaboration Sphere** | Proactively builds strong relationships with key contacts outside own area of expertise, Creates regular opportunities to collaborate strategically with functional senior staff and leadership. |

Reviewer: ```suggestion
| **Collaboration Sphere** | Proactively builds strong relationships with key contacts outside own area of expertise. Creates regular opportunities to collaborate strategically with functional senior staff and leadership. |
```

## Staff Site Reliability Engineer

<a name="staff_techexc"></a>
### Technical Excellence
###### [Jump to top](#main)
|||
|---|---|
| **Tools and Tech** | Experienced with all essential technologies for their role. Recognized as a company expert on particular tools/technologies. May contribute to technologies via bug reports/fixes or performance improvements when possible. Recognized as the company expert for the SRE. Mastered most of SRE concepts. |
| **Technical Problem Solving** | Works on and initiates technical projects that span multiple applications/products at Kensho. Solutions may involve coordinating deploys across many services. The scope of technical projects typically requires a collaborative interaction mode in contrast to x-as-a-service. Detects system- or project-level estimation and timeline issues. Notices bugs in cross-team behavior or ability to deliver. Improves velocity on large, months-long projects. Recognizes new technology necessary for Reliability, Observability and Scalability of all services and recognizes new technology necessary for model management and inference in production. (e.g. Collection of metrics, serving, storage and sync models). |

> | **Architecture** | Evaluates technology tradeoffs between complexity, memory usage, performance, reliability, and suitability for long term sustainability at Kensho. Articulates how those trade-offs manifest across teams and monitors suitability of team structures for the desired architecture(s). Understands architectures of all the running services and dependencies. Can suggest changes to a service to make it available in multi-region. Innovate and develop new tools/tech to eliminate most of the toil and reduce operation workload.  |

Reviewer: ```suggestion
| **Architecture** | Evaluates technology tradeoffs between complexity, memory usage, performance, reliability, and suitability for long term sustainability at Kensho. Articulates how those trade-offs manifest across teams and monitors suitability of team structures for the desired architecture(s). Understands architectures of all the running services and dependencies. Can suggest changes to a service to make it available in multi-region. Innovates and develops new tools/tech to eliminate most of the toil and reduce operation workload.  |
```

| **Implementation & Forethought** | Consistently builds software which is resilient to unexpected failures. Observability extends beyond project boundaries. Can adapt software to fit our needs. Carves a path for usage without disrupting any service. Takes actions based on error budgets. |
| **Code/Design Review** | Provides thorough, timely code reviews. Sought out as a reviewer for more complicated changes affecting multiple applications, or changes which are identified as requiring more thorough review. Provides design feedback on integrating applications into the broader Kensho ecosystem.  |
| **Collaboration** | Handles SRE migrations completely. Recognizes the need for re-reviews. Keeps the rest of the SRE team updated on changes to services, subtle implications and the effects on customers. Can migrate applications from Kensho to any S&P division. |

> | **Debugging & Support** | Debugs issues even in unfamiliar products with incomplete context. Offers mitigating or partial solutions; may drive and/or take ownership of solutions. Can reduce the duration and frequency of failures. Can identify and fix all of the production issues. Provides executive summary for all incidents to executives and customers. Own disaster management for all services. Creates environment to mock disasters and improve disaster recovery process regularly. |

Reviewer: ```suggestion
| **Debugging & Support** | Debugs issues even in unfamiliar products with incomplete context. Offers mitigating or partial solutions; may drive and/or take ownership of solutions. Can reduce the duration and frequency of failures. Can identify and fix all of the production issues. Provides executive summary for all incidents to executives and customers. Owns disaster management for all services. Creates environment to mock disasters and improve disaster recovery process regularly. |
```

| **Innovation** | Accustomed to the risks and benefits of failing and learning fast and does so frequently. Supports junior teammates through the learning process to achieve desired results.  |
| **Decision Scope** | Makes decisions that influence and drive the work of their peers and teammates. Plans always include multiple, well-thought out contingencies.  |
| **Operational Capacity** | Works on significant and unique issues where analysis of situations or data requires an evaluation of intangibles that is critical to the strategic planning and execution of important business processes.  |
| **Collaboration Sphere** | Creates formal networks involving coordination among groups and builds mostly strategic relationships with peers and company leadership. |
| **Conflict Resolution** | Addresses and resolves/diffuses conflict proactively, before escalation is required. Attuned at identifying conflict within own team; aids in resolving/diffusing promptly, professionally.  |

> | **Self-Development** | Demonstrative ability to take on responsibilities outside their immediate area of focus. Consistently creates opportunities to challenge themselves in new ways.  |

Reviewer: ```suggestion
| **Self-Development** | Demonstratives ability to take on responsibilities outside their immediate area of focus. Consistently creates opportunities to challenge themselves in new ways.  |
```

| **Adaptability** | Consistently reframes obstacles as opportunities. Solicits input from teams on ideas for mitigating obstacles and change. Seeks to understand historical decisions to avoid repeating mistakes and/or reassess decisions. |

> | **Judgement** | Exercises sound, independent judgement in all things, especially with methods, techniques and evaluation criteria for obtaining results. Work/business-related decisions often reflect the notion of putting company before self.  |

Reviewer: ```suggestion
| **Judgement** | Exercises sound, independent judgement in all things, especially with methods, techniques and evaluation criteria for obtaining results. Work/business-related decisions often reflect the notion of putting company before self. |
```

<a name="staff_autonomy"></a>
### Autonomy
###### [Jump to top](#main)
|||
|---|---|
| **Bias to Action** | Acts independently to determine methods and procedures on new or special assignments. May supervise the activities of others in some capacity. Makes swift decisions based on knowns, with consideration given to the unknowns, and accepts consequences of those decisions as long as desired outcomes are achieved.  |
| **Influence on Work** | Leverages own experience to successfully manage and direct cross team commitments and timelines. Influences work and team in a broad way, often at the departmental level.  |
| **Delegation** | Consistently provides broader context. Increases own technical leverage by delegating responsibility over outcomes, not tasks.  |
| **Prioritization** | Often determines the priorities and timelines of own and/or other teams. Understands and implements time effectiveness principles to keep self and others on track. |
| |  |

<a name="staff_teamwork"></a>
### Teamwork
###### [Jump to top](#main)
|||
|---|---|

> | **Feedback** | Is a primary source of feedback to peers, teammates, and managers. Versed and comfortable delivering constructive feedback. Understands which elements of feedback are critical to their own success; implements changes as needed.  |

Reviewer: ```suggestion
| **Feedback** | Is a primary source of feedback to peers, teammates, and managers. Versed and comfortable delivering constructive feedback. Understands which elements of feedback are critical to their own success; implements changes as needed. |
```
